core/src/error.rs

```diff
@@ -41,6 +41,14 @@ impl From<scale_decode::visitor::DecodeError> for Error {
     }
 }
 
+// TODO: when `codec::Error` implements `core::Error`
+// remove this impl and replace it by thiserror #[from]
+impl From<codec::Error> for Error {
+    fn from(err: codec::Error) -> Error {
+        Error::Codec(err)
+    }
+}
```

Comment:
Why does not implementing `core::Error` stop you from using `#[from]` here?

Reply:
```bash
error[E0599]: the method `as_dyn_error` exists for reference `&Error`, but its trait bounds were not satisfied
  --> core/src/error.rs:17:13
   |
17 |     Codec(#[from] codec::Error),
   |             ^^^^ method cannot be called on `&Error` due to unsatisfied trait bounds
   |
  ::: /home/niklasad1/.cargo/registry/src/index.crates.io-6f17d22bba15001f/parity-scale-codec-3.6.12/src/error.rs:26:1
   |
26 | pub struct Error {
   | ---------------- doesn't satisfy `parity_scale_codec::Error: AsDynError<'_>` or `parity_scale_codec::Error: StdError`
   |
   = note: the following trait bounds were not satisfied:
           `parity_scale_codec::Error: StdError`
           which is required by `parity_scale_codec::Error: AsDynError<'_>`
           `&parity_scale_codec::Error: StdError`
           which is required by `&parity_scale_codec::Error: AsDynError<'_>`
```

It's probably because thiserror implements core::error::Error and codec::Error implements std::error::Error but perhaps we could open issue in thiserror if there is some way around that